service/internal/graph/graph.go

```diff
@@ -45,12 +46,19 @@ type Graph struct {
 
 	// Keep track of how nodes relate to pipelines, so we can declare edges in the graph.
 	pipelines map[component.ID]*pipelineNodes
+
+	// Keep track of status source per node
+	instanceIDs map[int64]*component.InstanceID
+
+	telemetry servicetelemetry.TelemetrySettings
```

Comment:
why two servicetelemetry.TelemetrySettings fields?

Reply:
The `Settings` struct get passed to `Graph.Build` and is passed around from there, but goes out of scope after `Build`. I need access to `TelemetrySettings` later during `Graph.StartAll` and `Graph.ShutdownAll`. There needs to be a way to access the telemetry settings for the lifetime of `Graph`.

Reply:
Oh I see... I didn't realize the struct in question was Graph, and not Settings. Thanks